pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go

```diff
@@ -0,0 +1,209 @@
+package jobrunhistoricaldataanalyzer
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"os"
+	"time"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+)
+
+type JobRunHistoricalDataAnalyzerOptions struct {
+	ciDataClient jobrunaggregatorlib.CIDataClient
+	outputFile   string
+	newFile      string
+	currentFile  string
+	dataType     string
+	leeway       time.Duration
+}
+
+func (o *JobRunHistoricalDataAnalyzerOptions) Run(ctx context.Context) error {
+	newHistoricalData := []jobrunaggregatorapi.HistoricalDataRow{}
+
+	// We check what the current active release version is
+	currentRelease, previousRelease, err := fetchCurrentRelease()
+	if err != nil {
+		return err
+	}
+
+	currentHistoricalData, err := readHistoricalDataFile(o.currentFile)
+	if err != nil {
+		return err
+	}
+	if len(currentHistoricalData) == 0 {
+		return fmt.Errorf("current historical data is empty, can not compare")
+	}
+
+	switch {
+	case o.newFile == "" && o.dataType == "alerts":
+		newHistoricalData, err = o.ciDataClient.ListAlertHistoricalData(ctx)
+		if err != nil {
+			return err
+		}
+	case o.newFile == "" && o.dataType == "disruptions":
+		newHistoricalData, err = o.ciDataClient.ListDisruptionHistoricalData(ctx)
+		if err != nil {
+			return err
+		}
+	default:
+		newHistoricalData, err = readHistoricalDataFile(o.newFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(newHistoricalData) == 0 {
+		return fmt.Errorf("new historical data is empty, can not compare")
+	}
+
+	// We check to make sure the current data set doesn't contain the previous release version
+	// If that's the case, then we are in a new release event (i.e. a new branch has been cut)
+	// We then write to a `require_review` file to record why a review would be required.
+	currentVersionUpdate := currentDataContainsPreviousRelease(previousRelease, currentHistoricalData)
+	if currentVersionUpdate {
```

Comment:
This var name looks confusing, maybe newReleaseUpdate?

Reply:
That sounds good, I like that better, was struggling with naming this one